Errors reported by mypy 0.641

Newly-released mypy 0.641 issues these errors when checking Cranelift's code:
```
cdsl/ti.py:42: error: Too many arguments for "TypeConstraint"
cdsl/ti.py:78: warning: Returning Any from function declared to return "Iterable[TypeVar]"
```
I've [pinned](https://github.com/CraneStation/cranelift/commit/47e8d42f59070aacd07ed81d0fc41cca0fee7d1c) the build to 0.630 to unbreak the CI for now.

--- a/.travis.yml
+++ b/.travis.yml
@@ -23,9 +23,7 @@ addons:
         packages:
             - python3-pip
 install:
-    # Fix the mypy version at 0.630 since version 0.641 is more strict and the
-    # code is not yet updated.
-    - pip3 install --user --upgrade mypy==0.630 flake8
+    - pip3 install --user --upgrade mypy flake8
     - mypy --version
 before_script:
     # If an old version of rustfmt from cargo is already installed, uninstall

--- a/lib/codegen/meta-python/cdsl/ti.py
+++ b/lib/codegen/meta-python/cdsl/ti.py
@@ -26,6 +26,14 @@ class TypeConstraint(object):
     """
     Base class for all runtime-emittable type constraints.
     """
+
+    def __init__(self, tv, tc):
+        # type: (TypeVar, Union[TypeVar, TypeSet]) -> None
+        """
+        Abstract "constructor" for linters
+        """
+        assert False, "Abstract"
+
     def translate(self, m):
         # type: (Union[TypeEnv, TypeMap]) -> TypeConstraint
         """
@@ -75,7 +83,7 @@ class TypeConstraint(object):
         """
         Return the typevars contained in this constraint.
         """
-        return filter(lambda x:  isinstance(x, TypeVar), self._args())
+        return list(filter(lambda x: isinstance(x, TypeVar), self._args()))
 
     def is_trivial(self):
         # type: () -> bool

